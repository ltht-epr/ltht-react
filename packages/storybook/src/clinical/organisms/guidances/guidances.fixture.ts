import { Guidance, Metadata, GuidanceStatusCode } from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
  isRedacted: false,
  requestedWhen: '',
}

const redactedMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
  isRedacted: true,
  requestedWhen: '',
}

const guidanceOne: Guidance = {
  id: 'c2ee5d22-6fcd-4ec8-b12a-e9c6e83876b1',
  isEnteredInError: false,
  metadata: mockMetadata,
  text: {
    div: '',
    text:
      'As the information entered indicates that the patient has cognition and mental capacity issues, please consider the need for:',
  },
  note: [
    { text: 'Advice from MCA / MHA team' },
    { text: 'Advice from Psychiatric Liaison Team (if appropriate)' },
    { text: 'Assessment and Care of Psychological Health Care Plan' },
    { text: 'Enhanced care' },
    { text: 'Further assessment and / or referral to Mental Health team' },
    { text: 'Hospital passport' },
    { text: '‘Know Who I Am’ booklet (Forget Me Not)' },
    { text: 'Restrictive Intervention Care Plan' },
    { text: 'Self-Harm or Harm to Others Care Plan' },
    { text: "Utilising the 'Use of Restraint' policy and Good Practice Guidance" },
  ],
  status: GuidanceStatusCode.Success,
  reasonCode: {
    text: 'Risk - Communication',
  },
}

const guidanceTwo: Guidance = {
  id: 'c2ee5d22-6fcd-4ec8-b12a-e9c6e83876b2',
  isEnteredInError: false,
  metadata: redactedMetadata,
  text: {
    div: '',
    text: 'As the information entered indicates that the patient has breathing issues, please consider the need for:',
  },
  note: [{ text: 'Oxygen prescription' }, { text: 'Tracheostomy Care Plan"' }],
  status: GuidanceStatusCode.Success,
  reasonCode: {
    text: 'Risk - Cognition and Mental Capacity',
  },
}

const guidanceThree: Guidance = {
  id: '1e4fd41d-5c55-4c46-acc9-e216b3dc2732',
  isEnteredInError: false,
  metadata: mockMetadata,
  text: {
    div: '',
    text:
      "As you have indicated that the patient has hydration issues, please alert the Medical Team of the patient's AKI status",
  },
  status: GuidanceStatusCode.Success,
  reasonCode: {
    text: 'Risk - Unknown',
  },
}

const guidanceList: Guidance[] = [guidanceOne, guidanceTwo, guidanceThree]

// const guidanceList: GuidanceList = {
//   id: '88686ed1-ac86-4e14-8c73-fc020f1b0e5b',
//   metadata: mockMetadata,
//   guidance: [guidanceOne, guidanceTwo, guidanceThree],
// }

export default guidanceList
