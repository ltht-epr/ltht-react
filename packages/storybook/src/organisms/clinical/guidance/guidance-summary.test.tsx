import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import GuidanceSummary from '@ltht-react/guidance-summary'
import guidances from './guidance.fixture'

const Sut: React.FC = () => {
  return (
    <>
      {guidances.map(guidance => (
        <div key={guidance.id}>
          <GuidanceSummary guidance={guidance} />
        </div>
      ))}
    </>
  )
}

describe('GuidanceSummary', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Sut />, div)
  })
  it('matches snapshot', () => {
    expect(mount(<Sut />)).toMatchSnapshot('wrapper mount')
  })
})
