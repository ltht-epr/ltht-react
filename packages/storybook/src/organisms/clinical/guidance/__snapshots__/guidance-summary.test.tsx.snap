// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuidanceSummary matches snapshot: wrapper mount 1`] = `
.emotion-1 {
  display: block;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.emotion-2 {
  list-style-type: disc;
  list-style-position: inside;
}

.emotion-4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: left;
}

.emotion-3 {
  color: #768692;
}

.emotion-6 {
  display: block;
  padding-top: 0.5rem;
  padding-bottom: 0;
}

<Sut>
  <div
    key="c2ee5d22-6fcd-4ec8-b12a-e9c6e83876b1"
  >
    <GuidanceSummary
      guidance={
        Object {
          "id": "c2ee5d22-6fcd-4ec8-b12a-e9c6e83876b1",
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock",
              },
            ],
            "isRedacted": false,
            "requestedWhen": "",
          },
          "note": Array [
            Object {
              "text": "Advice from MCA / MHA team",
            },
            Object {
              "text": "Advice from Psychiatric Liaison Team (if appropriate)",
            },
            Object {
              "text": "Assessment and Care of Psychological Health Care Plan",
            },
            Object {
              "text": "Enhanced care",
            },
            Object {
              "text": "Further assessment and / or referral to Mental Health team",
            },
            Object {
              "text": "Hospital passport",
            },
            Object {
              "text": "‘Know Who I Am’ booklet (Forget Me Not)",
            },
            Object {
              "text": "Restrictive Intervention Care Plan",
            },
            Object {
              "text": "Self-Harm or Harm to Others Care Plan",
            },
            Object {
              "text": "Utilising the 'Use of Restraint' policy and Good Practice Guidance",
            },
          ],
          "reasonCode": Object {
            "text": "Risk - Communication",
          },
          "status": "SUCCESS",
          "text": Object {
            "div": "",
            "text": "As the information entered indicates that the patient has cognition and mental capacity issues, please consider the need for:",
          },
        }
      }
    >
      <Styled(h4)>
        <h4
          className="emotion-0"
        >
          Risk - Communication
        </h4>
      </Styled(h4)>
      <Styled(span)
        hasNotes={true}
      >
        <span
          className="emotion-1"
        >
          As the information entered indicates that the patient has cognition and mental capacity issues, please consider the need for:
        </span>
      </Styled(span)>
      <UnorderedList
        bullet="disc"
      >
        <ul
          className="list emotion-2"
        >
          <ListItem
            key="0"
          >
            <li
              className="list__item"
            >
              Advice from MCA / MHA team
            </li>
          </ListItem>
          <ListItem
            key="1"
          >
            <li
              className="list__item"
            >
              Advice from Psychiatric Liaison Team (if appropriate)
            </li>
          </ListItem>
          <ListItem
            key="2"
          >
            <li
              className="list__item"
            >
              Assessment and Care of Psychological Health Care Plan
            </li>
          </ListItem>
          <ListItem
            key="3"
          >
            <li
              className="list__item"
            >
              Enhanced care
            </li>
          </ListItem>
          <ListItem
            key="4"
          >
            <li
              className="list__item"
            >
              Further assessment and / or referral to Mental Health team
            </li>
          </ListItem>
          <ListItem
            key="5"
          >
            <li
              className="list__item"
            >
              Hospital passport
            </li>
          </ListItem>
          <ListItem
            key="6"
          >
            <li
              className="list__item"
            >
              ‘Know Who I Am’ booklet (Forget Me Not)
            </li>
          </ListItem>
          <ListItem
            key="7"
          >
            <li
              className="list__item"
            >
              Restrictive Intervention Care Plan
            </li>
          </ListItem>
          <ListItem
            key="8"
          >
            <li
              className="list__item"
            >
              Self-Harm or Harm to Others Care Plan
            </li>
          </ListItem>
          <ListItem
            key="9"
          >
            <li
              className="list__item"
            >
              Utilising the 'Use of Restraint' policy and Good Practice Guidance
            </li>
          </ListItem>
        </ul>
      </UnorderedList>
    </GuidanceSummary>
  </div>
  <div
    key="c2ee5d22-6fcd-4ec8-b12a-e9c6e83876b2"
  >
    <GuidanceSummary
      guidance={
        Object {
          "id": "c2ee5d22-6fcd-4ec8-b12a-e9c6e83876b2",
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock",
              },
            ],
            "isRedacted": true,
            "requestedWhen": "",
          },
          "note": Array [
            Object {
              "text": "Oxygen prescription",
            },
            Object {
              "text": "Tracheostomy Care Plan\\"",
            },
          ],
          "reasonCode": Object {
            "text": "Risk - Cognition and Mental Capacity",
          },
          "status": "SUCCESS",
          "text": Object {
            "div": "",
            "text": "As the information entered indicates that the patient has breathing issues, please consider the need for:",
          },
        }
      }
    >
      <GuidanceRedacted>
        <div
          className="emotion-4"
        >
          <RedactedDescription>
            <div
              className="emotion-3"
            >
              Insufficient privileges
            </div>
          </RedactedDescription>
        </div>
      </GuidanceRedacted>
    </GuidanceSummary>
  </div>
  <div
    key="1e4fd41d-5c55-4c46-acc9-e216b3dc2732"
  >
    <GuidanceSummary
      guidance={
        Object {
          "id": "1e4fd41d-5c55-4c46-acc9-e216b3dc2732",
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock",
              },
            ],
            "isRedacted": false,
            "requestedWhen": "",
          },
          "reasonCode": Object {
            "text": "Risk - Unknown",
          },
          "status": "SUCCESS",
          "text": Object {
            "div": "",
            "text": "As you have indicated that the patient has hydration issues, please alert the Medical Team of the patient's AKI status",
          },
        }
      }
    >
      <Styled(h4)>
        <h4
          className="emotion-0"
        >
          Risk - Unknown
        </h4>
      </Styled(h4)>
      <Styled(span)
        hasNotes={false}
      >
        <span
          className="emotion-6"
        >
          As you have indicated that the patient has hydration issues, please alert the Medical Team of the patient's AKI status
        </span>
      </Styled(span)>
      <UnorderedList
        bullet="disc"
      >
        <ul
          className="list emotion-2"
        />
      </UnorderedList>
    </GuidanceSummary>
  </div>
</Sut>
`;
